File: python/longest-common-prefix/longest_common_prefix.py
from pip import List


class LongestCommonPrefix:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        if len(strs) == 1:
            return strs[0]

        shortest = min(strs, key=len)
        for i, ch in enumerate(shortest):
            for string in strs:
                if string[i] != ch:
                    return string[0:i]
        return shortest

File: typescript/longest-common-prefix/src/longest-common-prefix.ts
function longestCommonPrefix(strs: string[]): string {
  if (strs.length == 0) {
    return "";
  }
  if (strs.length == 1) {
    return strs[0];
  }
  const smallest = strs.reduce((a, b) => {
    if (a.length > b.length) {
      return a;
    }
    return b;
  });

  for (let i = 0; i < smallest.length; i++) {
    if (
      !strs.every((str) => {
        return str[i] == smallest[i];
      })
    ) {
      return smallest.substring(0, i);
    }
  }

  return smallest;
}

export default longestCommonPrefix;
